src/vm/i386/asmhelpers.S

```diff
@@ -1162,20 +1162,30 @@ NESTED_END ResolveWorkerChainLookupAsmStub, _TEXT
 NESTED_ENTRY BackPatchWorkerAsmStub, _TEXT, NoHandler
     PROLOG_BEG
     PROLOG_PUSH eax // it may contain siteAddrForRegisterIndirect
+    PROLOG_PUSH ebx
     PROLOG_PUSH ecx
     PROLOG_PUSH edx
     PROLOG_END
 
-    sub     esp, 4     //  for 16 bytes align
+    // Align Stack Frame
+    mov     ebx, esp
+    and     esp, -16
+
+    // Call BackPatchWorkerStaticStub
+    sub     esp, 8     //  16-bytes align padding
```

Comment:
@parjong I am sorry, I really meant the frame above the stub that's the managed code. You can see the stub code layout in the source that you've referred to. As you can see in the code, there are no pushes, just the call to the BackPatchWorkerAsmStub. And the stub is called from the managed code with the parameters corresponding to the virtual call plus possibly eax set to siteAddrForRegisterIndirect. So the JIT should have aligned the stack before the call properly. At the entry to the stub, the esp would be the managed caller esp - 4, so when it arrives in the BackPatchWorkerAsmStub, the esp would be the managed caller esp - 8. So it seems the BackPatchWorkerAsmStub should add padding of 8.